### PR TITLE
CI: enable publishing of documentation on pages

### PR DESCRIPTION
**Pull request by leeks-int**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,7 +55,6 @@ jobs:
   documetation:
```
Comment (zheltovs) [inline .github/workflows/ci.yml:55]: `documetation` -> `documentation`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,16 +64,46 @@ jobs:
...
+          export BUILD_TYPE=${BUILD_TYPE} && sphinx-build doc/main html
+          tar -czvf html.tar.gz html/
       - name: Save docs
         uses: actions/upload-artifact@v2.2.1
         with:
           name: oneTBB-html-docs-${{ env.GITHUB_SHA_SHORT }}
           path: html
 
+  pages:
+    if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (AlexVeprev) [inline .github/workflows/ci.yml:79]: `main` -> `master` ?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (leeks-int) [reply .github/workflows/ci.yml]: Updated

File: .github/workflows/ci.yml (hunk 2 above)
Comment (leeks-int) [reply .github/workflows/ci.yml]: Fixed

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -65,16 +64,46 @@ jobs:
...
 
+  pages:
+    if: ${{ github.ref == 'refs/heads/master' }}
+    runs-on: ubuntu-latest
+    needs: [documentation]
+    steps:
+      - name: Checkout gh-pages
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
```
Comment (AlexVeprev) [inline .github/workflows/ci.yml:86]: Make sure you create orphan branch `gh-pages` in the repo before merging the PR.

**Review by AlexVeprev — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (leeks-int) [reply .github/workflows/ci.yml:86]: Created

**Pull request merged**